source/common/router/shadow_writer_impl.cc

```diff
@@ -58,7 +58,12 @@ Upstream::ThreadLocalCluster* ShadowWriterImpl::getClusterAndPreprocessHeadersAn
     return nullptr;
   }
 
-  headers.setHost(shadowAppendedHost(headers.getHostValue()));
+  // Append "-shadow" suffix if option is disabled.
+  if (const_cast<Http::AsyncClient::RequestOptions&>(options).is_shadow_suffixed_disabled) {
+    headers.setHost(headers.getHostValue());
```

Comment:
Why is this line needed?
IIUC it just does: `headers.host = headers.host;`, no?